### PR TITLE
Prevent remote dashboard auto-open

### PR DESCRIPTION
**Pull request by ttupper92618**
## Summary

- restrict first-run dashboard auto-open to a local interactive terminal
- suppress GUI browser launches from SSH, redirected, service, and native-app runtimes while still printing the dashboard URL
- retain the first-run marker even when auto-open is intentionally skipped
- document the startup behavior and add focused coverage for every launch context

## Root cause

Fresh-install qualification creates a new empty home on each run. Skulk treated
each run as a first launch and called `webbrowser.open`, which macOS routed into
the logged-in desktop Chrome session on the remote target. Repeated
qualification retries therefore accumulated live dashboard tabs and renderer
processes.

## User impact

Local users starting Skulk interactively still get the convenient first-run
browser open. Remote nodes, services, redirected commands, and automated
qualification runs no longer modify a logged-in user's GUI browser.

## Validation

- `uv run basedpyright`
- `uv run ruff check`
- `nix --extra-experimental-features 'nix-command flakes' fmt`
- `uv run pytest` (`2966 passed, 1 skipped, 228 deselected`)

### CODE REVIEW

File: src/skulk/utils/banner.py [hunk 1]
```diff
@@ -19,7 +20,34 @@ def _mark_first_run_done() -> None:
...
+    commands and non-interactive service or qualification launches can still
+    have access to a logged-in macOS GUI session, where ``webbrowser.open``
+    would otherwise accumulate tabs in the user's desktop browser.
+    """
+
+    return (
+        not os.environ.get("SKULK_RUNTIME_DIR")
+        and not any(os.environ.get(name) for name in _SSH_ENVIRONMENT_VARIABLES)
+        and sys.stderr.isatty()
+    )
```
[the remark this answers is not in the document]
Comment (ttupper92618) [reply src/skulk/utils/banner.py:36]: Severity 2 (low): a normal detached stderr stream still implements `isatty()` and returns false. A runtime that sets `sys.stderr = None` would already fail at the pre-existing `print(..., file=sys.stderr)` before this new gate, so guarding only this call would not make that runtime safe. A custom printable stream without `isatty()` is a narrow injected edge case rather than the launch paths this PR corrects; no code change in this PR.

File: src/skulk/utils/tests/test_banner.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+from __future__ import annotations
+
+import io
+from pathlib import Path
+
+import pytest
+
+from skulk.utils import banner
+
+_SSH_ENVIRONMENT_VARIABLES = ("SSH_CLIENT", "SSH_CONNECTION", "SSH_TTY")
```
[the remark this answers is not in the document]
Comment (ttupper92618) [reply src/skulk/utils/tests/test_banner.py:12]: Severity 1 (informational): the test intentionally enumerates the three standard SSH variables as the external behavior contract. Importing the production module's private constant would couple the test to implementation and violate this repository's strict private-usage checking. The parameterized test already proves each supported variable independently; no change.
